editor/components/post-taxonomies/flat-term-selector.js

```diff
@@ -200,18 +206,15 @@ class FlatTermSelector extends Component {
 }
 
 export default compose(
-	withAPIData( ( props ) => {
-		const { slug } = props;
-		return {
-			taxonomy: `/wp/v2/taxonomies/${ slug }?context=edit`,
-		};
-	} ),
-	withSelect( ( select, ownProps ) => {
+	withSelect( ( select, { slug } ) => {
 		const { getCurrentPost } = select( 'core/editor' );
+		const { getTaxonomy } = select( 'core' );
+		const taxonomy = getTaxonomy( slug );
 		return {
-			hasCreateAction: get( getCurrentPost(), [ '_links', 'wp:action-create-' + ownProps.restBase ], false ),
-			hasAssignAction: get( getCurrentPost(), [ '_links', 'wp:action-assign-' + ownProps.restBase ], false ),
-			terms: select( 'core/editor' ).getEditedPostAttribute( ownProps.restBase ),
+			hasCreateAction: taxonomy ? get( getCurrentPost(), [ '_links', 'wp:action-create-' + taxonomy.rest_base ], false ) : false,
+			hasAssignAction: taxonomy ? get( getCurrentPost(), [ '_links', 'wp:action-assign-' + taxonomy.rest_base ], false ) : false,
+			terms: taxonomy ? select( 'core/editor' ).getEditedPostAttribute( taxonomy.rest_base ) : [],
+			taxonomy,
```

Comment:
Actually, this made me think, what will happen when `taxonomy` is falsy? We don't check against it inside the component.

Reply:
We check against `hasAssignAction` which is false if the taxonomy is falsy.

Reply:
And the taxonomy object is used for labels as well.

Reply:
Do we expect to refactor this component so that it's not calling `apiFetch` at all? This was a hope of mine. Wondering if it would make things easier to do this conversion first, before this one.

Reply:
Yes, I think we should do it, the thing is, it's not ready at the moment because we need paginated/query data in `core-data`. Hopefully we can get back to that once we publish those remaining packages.